src/js/components/sidebar.jsx

```diff
@@ -293,12 +336,22 @@ export default class SideBar extends React.Component {
   }
 
   _onSelectionChanged(event, tabId) {
-    const { selectAll, tabsById } = this.state;
+    const { tabsById, filteredTabs, selectAll } = this.state;
+
     tabsById.get(tabId).selected = event.target.checked;
 
-    this.setState(
-      selectAll ? { ...this.state, selectAll: false } : { ...this.state },
-    );
+    const hasUnselectedTabs =
+      filteredTabs.find(id => !tabsById.get(id).selected) == undefined
+        ? false
+        : true;
+
+    if (!hasUnselectedTabs) {
```

Comment:
There's some boolean flipping that my brain has to do when I think "not has non-selected tabs", so I think this could be simplified... I suppose we could have `const hasSelectedTabs` instead - that might make this clearer.